specifyweb/frontend/js_src/lib/components/FormParse/index.ts

```diff
@@ -119,63 +119,82 @@ export const fetchView = async (
         return views[name];
       });
 
-export function parseViewDefinition(
+export async function parseViewDefinition(
   view: ViewDefinition,
   defaultType: FormType,
   originalMode: FormMode,
   currentTable: SpecifyTable
-): ViewDescription | undefined {
+): Promise<ViewDescription | undefined> {
   const logContext = getLogContext();
   addContext({ view, defaultType, originalMode });
 
-  const resolved = resolveViewDefinition(view, defaultType, originalMode);
-  if (resolved === undefined) return undefined;
-  addContext({ resolved });
-  const { mode, formType, viewDefinition, table = currentTable } = resolved;
-
-  const parser =
-    formType === 'formTable'
-      ? parseFormTableDefinition
-      : (
-          viewDefinition: SimpleXmlNode,
-          table: SpecifyTable
-        ): ParsedFormDefinition =>
-          parseFormDefinition(viewDefinition, table)[0].definition;
-
-  const [errors, parsed] = captureLogOutput(() =>
-    parser(viewDefinition, table)
+  return resolveViewDefinition(view, defaultType, originalMode).then(
+    (resolved) => {
+      if (resolved === undefined) return undefined;
+      addContext({ resolved });
+      const { mode, formType, viewDefinition, table = currentTable } = resolved;
+
+      const parser =
+        formType === 'formTable'
+          ? parseFormTableDefinition
+          : (
+              viewDefinition: SimpleXmlNode,
+              table: SpecifyTable
+            ): ParsedFormDefinition =>
+              parseFormDefinition(viewDefinition, table)[0].definition;
+
+      const [errors, parsed] = captureLogOutput(() =>
+        parser(viewDefinition, table)
+      );
+      setLogContext(logContext);
+
+      return {
+        mode,
+        formType,
+        table,
+        viewSetId: view.viewsetId ?? undefined,
+        errors,
+        name: view.name,
+        ...parsed,
+      };
+    }
   );
-  setLogContext(logContext);
-
-  return {
-    mode,
-    formType,
-    table,
-    viewSetId: view.viewsetId ?? undefined,
-    errors,
-    name: view.name,
-    ...parsed,
-  };
 }
 
-export function resolveViewDefinition(
+export async function resolveViewDefinition(
   view: ViewDefinition,
   formType: FormType,
   mode: FormMode
-):
+): Promise<
   | {
       readonly viewDefinition: SimpleXmlNode;
       readonly formType: FormType;
       readonly mode: FormMode;
       readonly table: SpecifyTable | undefined;
     }
-  | undefined {
+  | undefined
+> {
   const viewDefinitions = parseViewDefinitions(view.viewdefs);
   if (Object.keys(viewDefinitions).length === 0) {
     console.error(`No view definitions found for the ${view.name} view`);
     return undefined;
   }
 
+  const useDefaultExpanded =
+    formType === 'form' &&
+    new Set(Object.values(view.altviews).map((altView) => altView.viewdef))
+      .size === 1 &&
+    (toSimpleXmlNode(xmlToJson(viewDefinitions[view.altviews[0].viewdef]))
+      .attributes.type as Lowercase<FormType>) === 'formtable';
```

Comment:
given that you only access attributes, you don't need toSimpleXmlNode here

you can also use getAttribute:

```suggestion
    getAttribute(xmlToJson(viewDefinitions[view.altviews[0].viewdef],'type') === 'formtable';
```